src/main/java/stellarium/DrawSky.java

```diff
@@ -295,8 +300,8 @@ else if (mc.theWorld.provider.isSurfaceWorld())
 				difm.set(VecMath.mult(sizem, difm));
 				difm2.set(VecMath.mult(sizem, difm2));
 
-				float alpha=Optics.getAlphaFromMagnitude(-17.0-StellarSky.getManager().Moon.mag-2.5*Math.log10(difactor),bglight);
-
+				float alpha=(float) (Optics.getAlphaFromMagnitude(-17.0-StellarSky.getManager().Moon.mag-2.5*Math.log10(difactor),bglight) / (StellarSky.getManager().Moon.getPhase()));		
```

Comment:
This is for the Halo texture. A new moon should not have a halo, a full moon should have a full brightness halo, and a half moon's halo should be half-bright. I used the moon's phase to divide the alpha.